File: src/main/java/com/baulsupp/oksocial/okhttp/OkHttpResponseFuture.java
package com.baulsupp.oksocial.okhttp;

import java.io.IOException;
import java.util.concurrent.CompletableFuture;
import okhttp3.Call;
import okhttp3.Callback;
import okhttp3.Response;

public class OkHttpResponseFuture implements Callback {
  public final CompletableFuture<Response> future = new CompletableFuture<>();
  public Call call;

  public OkHttpResponseFuture(Call call) {
    this.call = call;
  }

  @Override public void onFailure(Call call, IOException e) {
    future.completeExceptionally(e);
  }

  @Override public void onResponse(Call call, Response response) throws IOException {
    future.complete(response);
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/okhttp/OkHttpResponseFuture.kt
package com.baulsupp.oksocial.okhttp

import okhttp3.Call
import okhttp3.Callback
import okhttp3.Response
import java.io.IOException
import java.util.concurrent.CompletableFuture

class OkHttpResponseFuture : Callback {
  val future = CompletableFuture<Response>()

  override fun onFailure(call: Call, e: IOException) {
    future.completeExceptionally(e)
  }

  override fun onResponse(call: Call, response: Response) {
    future.complete(response)
  }
}
